content/en/docs/concepts/workloads/pods/pod-lifecycle.md

```diff
@@ -460,14 +460,21 @@ An example flow:
       The containers in the Pod receive the TERM signal at different times and in an arbitrary
       order. If the order of shutdowns matters, consider using a `preStop` hook to synchronize.
       {{< /note >}}
-1. At the same time as the kubelet is starting graceful shutdown, the control plane removes that
-   shutting-down Pod from Endpoints (and, if enabled, EndpointSlice) objects where these represent
+
+{{< feature-state for_k8s_version="v1.22" state="beta" >}}
+
+1. At the same time as the kubelet is starting graceful shutdown, the control plane evaluates whether
+   shutting-down Pod needs to be removed from Endpoints (and, if enabled, EndpointSlice) objects where these represent
    a {{< glossary_tooltip term_id="service" text="Service" >}} with a configured
    {{< glossary_tooltip text="selector" term_id="selector" >}}.
    {{< glossary_tooltip text="ReplicaSets" term_id="replica-set" >}} and other workload resources
    no longer treat the shutting-down Pod as a valid, in-service replica. Pods that shut down slowly
-   cannot continue to serve traffic as load balancers (like the service proxy) remove the Pod from
-   the list of endpoints as soon as the termination grace period _begins_.
+   should not continue to serve traffic as normal and should be terminating connection. Without feature
+   gate `EndpointSliceTerminatingCondition` (in beta starting Kubernetes version 1.22), load balancers (like the service proxy) remove the Pod from
+   the list of endpoints as soon as the termination grace period _begins_. With the feature
+   gate `EndpointSliceTerminatingCondition`, Endpoints are not removed for the terminating pods
+   and new status indication terminating state is exposed from Endpoint API.
```

Comment:
Just so it's not confused with old Endpoint API, I would replace all Endpoints with EndpointSlice here. `Endpoint` API does not (and will not) support inclusion of terminating endpoints.

Reply:
A reference to the [EndpointSlice conditions documentation]( https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#conditions) here would be helpful I think, because for backwards compatibly a terminating endpoint is always Ready=True, so if you care about the actual readiness of a terminating pod, you need to check a new condition `serving`.

Reply:
Updated